instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -31,7 +31,7 @@ def trace_response(env, response)
             sinatra_response = ::Sinatra::Response.new([], response.first)
             return unless sinatra_response.server_error?
 
-            span.record_exception(env['sinatra.error'])
+            span.record_exception(env['sinatra.error']) if env['sinatra.error']
```

Comment:
Would you kindly add an automated test that we may use to avoid a future regression?

Reply:
I've been digging for how to remove/make-nil `env['sinatra.error']` for a while and can't seem to find a good example of how to intercept calls to that variable that is passed into `call`. If you have any tips or places to look lmk, thank you!

Reply:
Hmmm... how does it manifest itself in production? Does the Sinatra code base have a test case that covers this that you may use for inspiration?

Reply:
We have a custom error handler class that that calls [Sinatra's Base error](https://www.rubydoc.info/gems/sinatra/1.4.6/Sinatra%2FBase.error), and then calls:
```
ensure
    env.delete("sinatra.error")
end
```
I can't put all of it here b/c it's our closed source code unfortunately. 

I have a few options:

* Add some new mocking libraries that allow me to intercept all calls to Hashes and when `sinatra.error` is accessed, return nil. 
* Figure out if some middleware can delete that key (tried this already, can't seem to make it work)
* Call `trace_response` with mock/stubbed `env` and `response` input and verify it works when `env` is missing `sinatra.error`
* Intercept calls to `TracerMiddleware#trace_response`, remove `sinatra.error` and then call the original function. I like this option, but can't figure out how to make it work.
* ?

Thoughts?

Reply:
> Call trace_response with mock/stubbed env and response input and verify it works when env is missing sinatra.error

This might be the easiest way.

Reply:
@ahayworth should we merge to get past the bug and ask @rdoherty-fastly to follow up with test coverage?

Reply:
I would be okay with releasing without the test case. A follow-on PR for test coverage would be fine with me.